src/cascadia/QueryExtension/OpenAILLMProvider.cpp

```diff
@@ -24,12 +24,21 @@ static constexpr std::wstring_view openAIEndpoint{ L"https://api.openai.com/v1/c
 
 namespace winrt::Microsoft::Terminal::Query::Extension::implementation
 {
-    void OpenAILLMProvider::SetAuthentication(const Windows::Foundation::Collections::ValueSet& authValues)
+    void OpenAILLMProvider::SetAuthentication(const winrt::hstring& authValues)
     {
-        _AIKey = unbox_value_or<hstring>(authValues.TryLookup(L"key").try_as<IPropertyValue>(), L"");
         _httpClient = winrt::Windows::Web::Http::HttpClient{};
         _httpClient.DefaultRequestHeaders().Accept().TryParseAdd(applicationJson);
-        _httpClient.DefaultRequestHeaders().Authorization(WWH::Headers::HttpCredentialsHeaderValue{ L"Bearer", _AIKey });
+
+        if (!authValues.empty())
+        {
+            // Parse out the key from the authValues string
+            WDJ::JsonObject authValuesObject{ WDJ::JsonObject::Parse(authValues) };
```

Comment:
this isn't gonna throw for people who had an old format key stored, right

Reply:
See above!